src/platform/Ameba/KeyValueStoreManagerImpl.cpp

```diff
@@ -94,7 +94,7 @@ CHIP_ERROR KeyValueStoreManagerImpl::_Put(const char * key, const void * value,
     if (checkExist(key, key))
     {
         ret = deleteKey(key, key);
-        if (TRUE != ret)
+        if (ret != 0)
```

Comment:
Consider creating a separate variable with a better name than `ret`. In this case something like `errorCode` would have made it obvious that a code cannot be compared with a bool.

Same for `ret` used at line 103 below. Name it `success` or similar, then it is more clearly a TRUE/FALSE. also if you use true/false TRUE is not a great compare as `truthy` applies to all non-zero (say 2,3 and very often used `-1`). I would recommend comparing with FALSE which is only one value (0).

Giving checkmark anyway because platform-specific code. Thinking that maintainability of this code could be improved though.

Reply:
Hi @andy31415, we are working on implementing error mapping to map ameba sdk errors to chip errors. We will use a separate PR for this. Can we merge this first? I have created an issue to track this #25912.

Reply:
Resolving, pls reopen if needed